src/Components/AllIngredientsHandler.cs

```diff
@@ -42,173 +36,342 @@ public AllIngredientsHandler(IBackend backend, IUser user)
+                Text = "Search ingredients...",
+                FontSize = boxFont,
+                BorderThickness = new Thickness(0),
+                Margin = new Thickness(10, 0, 0 , 0)
             };
 
-            // Clear placeholder on focus
+            // Clear placeholder text when the box is focused
             searchBox.GotFocus += (s, e) =>
             {
-                if (searchBox.Text == "Search ingredients...") searchBox.Text = string.Empty;
+                if (searchBox.Text == "Search ingredients...")
+                {
+                    searchBox.Text = string.Empty;
+                }
             };
-            searchBox.LostFocus += (s, e) =>
+
+            // Restore placeholder text if the box is empty when focus is lost
+            searchBox.LostFocus += async (s, e) =>
             {
-                if (string.IsNullOrEmpty(searchBox.Text)) searchBox.Text = "Search ingredients...";
+                if (string.IsNullOrEmpty(searchBox.Text))
+                {
+                    searchBox.Text = "Search ingredients...";
+                    await DisplayIngredientsAsync(parentPanel);
+                }
             };
 
-            return searchBox;
+            searchBoxBorder.Child = searchBox;
+
+            return searchBoxBorder;
         }
 
+
         /// <summary>
         /// Creates a row with ingredient details and an add button, wrapped in a border.
         /// </summary>
         /// <param name="ingredient">The ingredient to display in the row.</param>
         /// <returns>A styled Border containing the ingredient row.</returns>
         private Border CreateIngredientRow(Ingredient ingredient)
         {
-            DockPanel ingredientRow = new DockPanel { Margin = new Thickness(0, 5, 0, 5) };
-            TextBlock ingredientText = new TextBlock
+            // Define resource-based colors and font sizes
+            SolidColorBrush boxTextCol = (SolidColorBrush)App.Current.Resources["SecondaryBrushB"];
+            SolidColorBrush boxBorderCol = (SolidColorBrush)App.Current.Resources["SecondaryBrushB"];
+            int ingNameFont = 28;
+            int ingTypeFont = 26;
+
+            int boxButtonFont = 40;
+
+            // Create ingredient row 
+            DockPanel ingredientRow = CreateIngredientRowPanel();
+
+            // Create a container for the text blocks
+            StackPanel textContainer = new StackPanel
             {
-                Text = $"{ingredient.GetName()} - {ingredient.GetIngType()}",
-                Foreground = ingredientTextColor,
+                Orientation = Orientation.Vertical,
+                MaxWidth = parentPanel.ActualWidth / 4,
+                HorizontalAlignment = HorizontalAlignment.Left
+            };
+
+            // Create and add the ingredient name TextBlock
+            TextBlock ingredientNameText = CreateTextBlock($"{ingredient.GetName()}", boxTextCol, ingNameFont, FontWeights.Bold);
```

Comment:
Make the ingredient text wrap around for long texts. Right now, the name is just cutoff.
![image](https://github.com/user-attachments/assets/c57aa323-6bf0-4b37-a646-2a099ba7a166)

Reply:
By wrap around, you mean on the other line?. I'm not so sure if I can do that without some major changes to how the ingredient boxes are made. They are a fixed height and width. So doing this may make the rest of the ingredient boxes (the ones with normal length names) look a bit empty.

Reply:
What if you wrap the text around but if it overflows, make the box scrollable?